### PR TITLE
Update merge-queue gated feature for GHES 3.12

### PR DESCRIPTION
**Pull request by willsmythe**
### Why:

The "Who can use this feature" section for Merge Queue related docs (when viewing GHES 3.12) is confusing since it currently only mentions Enterprise Cloud. For example:

https://docs.github.com/en/enterprise-server@3.12/repositories/configuring-branches-and-merges-in-your-repository/configuring-pull-request-merges/managing-a-merge-queue

This change updates the gated feature details to be correct for GHES.

Note: the MQ feature is currently gated to GHES 3.12 and later.

### What's being changed (if available, include any code snippets, screenshots, or gifs):

Fixes the "Who can use this feature" section on Merge Queue related pages when viewing GHES 3.12 and later docs.

### Check off the following:

- [x] I have reviewed my changes in staging, available via the **View deployment** link in this PR's timeline (this link will be available after opening the PR).

  - For content changes, you will also see an automatically generated comment with links directly to pages you've modified. The comment won't appear if your PR only edits files in the `data` directory.
- [x] For content changes, I have completed the [self-review checklist](https://docs.github.com/en/contributing/collaborating-on-github-docs/self-review-checklist).

### CODE REVIEW

**Comment by willsmythe** [comment]
Verified the correct message appears for Cloud (i.e. the current message):
https://docs-31946-d0a842.preview.ghdocs.com/en/pull-requests/collaborating-with-pull-requests/incorporating-changes-from-a-pull-request/merging-a-pull-request-with-a-merge-queue

And the updated message appears for GHES 3.12:
https://docs-31946-d0a842.preview.ghdocs.com/en/enterprise-server@3.12/pull-requests/collaborating-with-pull-requests/incorporating-changes-from-a-pull-request/merging-a-pull-request-with-a-merge-queue

File: data/reusables/gated-features/merge-queue.md [hunk 1]
```diff
@@ -1 +1,5 @@
+{%- ifversion fpt or ghec %}
 Pull request merge queues are available in any public repository owned by an organization, or in private repositories owned by organizations using {% data variables.product.prodname_ghe_cloud %}. {% data reusables.gated-features.more-info %}
+{%- elsif ghes %}
+Pull request merge queues are available in any organization-owned repository in {% data variables.product.product_name %}.
```
Comment (isaacmbrown) [inline data/reusables/gated-features/merge-queue.md:4]:
```suggestion
Pull request merge queues are available in any organization-owned repository on {% data variables.product.product_name %}.
```

File: data/reusables/gated-features/merge-queue.md [hunk 2]
```diff
@@ -1 +1,5 @@
+{%- ifversion fpt or ghec %}
 Pull request merge queues are available in any public repository owned by an organization, or in private repositories owned by organizations using {% data variables.product.prodname_ghe_cloud %}. {% data reusables.gated-features.more-info %}
+{%- elsif ghes %}
```
Comment (isaacmbrown) [inline data/reusables/gated-features/merge-queue.md:3]:
```suggestion
{% ifversion fpt or ghec %}
Pull request merge queues are available in any public repository owned by an organization, or in private repositories owned by organizations using {% data variables.product.prodname_ghe_cloud %}. {% data reusables.gated-features.more-info %}
{% elsif ghes %}
```

**Review by isaacmbrown — APPROVED** [review]
Looks good, thanks @willsmythe 🙇

**Pull request merged**